### PR TITLE
Ensure the build check fails if there are new untracked files

### PR DESCRIPTION
**Pull request by brrygrdn**
One gap in the current check run is that if a PR generates _new files_ during a build which aren't checked as part of the PR, we ignore this.

This at least creates confusion as to why there is are untracked files locally when building on `main`, but in the worst case could cause us to fail to check in something that might be required in certain runtime environments.

## Ignoring .node files generated by ssh2

We pull in ssh2 via dockerode:

```
dockerode
↳ docker-modem
  ↳ ssh2
    ↳ cpu-features
```

Both it and `cpu-features` compile _optional_ native extensions using
OpenSSH and cpu-features (https://github.com/google/cpu_features) which
result in node-gyp generating two `.node` executables for these bindings

We have found that these bindings are very sensitive to the build env
which means that developer laptops, Actions and Codespaces result in
a diff in the `.node` files, something we seek to prevent in PRs in
order to detect cases where code changes are committed without actually
being applied to the `dist/` folder.

Since we do not actually SSH into any containers in our implementation,
let's just ignore these files in our distributed code rather than
make our build more convoluted/less portable.

### Why not just pin OpenSSH?

That was my first thought, but both Codespaces and Actions appear to be using the same base image of Ubuntu and OpenSSH/OpenSSL:

> ssh -V
> OpenSSH_8.2p1 Ubuntu-4ubuntu0.4, OpenSSL 1.1.1f  31 Mar 2020

I didn't want to get much further into the weeds of checking which developer lib versions where on both as the action currently works without these files being distributed, so they aren't involved in code paths we actually call.

### CODE REVIEW

**Comment by brrygrdn** [comment]
My expectation is that `check dist/ verify build` will fail as we currently have some untracked files on `main`'s build. I'll add these in a follow-up commit so we have a clear red > green signal this is resolving an issue.

**Review by jurre — APPROVED** [review]

File: script/check-diff [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+# Make sure we notice any untracked files generated by the build
+git add --intent-to-add .
```
Comment (landongrindheim) [inline script/check-diff:4]: This creeps outside the scope of the change you're introducing, but is related:

What do you think about replacing `--quiet` with `--exit-code`? My thinking is that we'd still trigger a build failure if the contents are different, but we'd be able to inspect the diff.

**Review by landongrindheim — APPROVED** [review]
Had a thought that I felt was very related to this change, but not within the scope of what you're doing. Feel free to ignore it 😄

File: script/check-diff (hunk 1 above)
Comment (brrygrdn) [reply script/check-diff:4]: That's a good call, I've modified the script to use `--quiet`, but in the event of a failure, output the full diff on line 8 (not in diff)

**Pull request merged**